parlai/core/logging_utils.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import sys
+import logging
+from pathlib import Path
+from datetime import date
+
+INFO = logging.INFO
+DEBUG = logging.DEBUG
+WARN_LEVEL = logging.WARNING
+ERROR = logging.ERROR
+CRITICAL = logging.CRITICAL
+
+
+class ParlaiLogger(logging.Logger):
+    def __init__(
+        self,
+        name,
+        console_level,
+        console_format=None,
+        file_format=None,
+        file_level=INFO,
+        filename=None,
+    ):
+        """
+        Initialize the logger object
+        :param name: Name of the logger
+        :param console_level: min. Level of messages logged to console
+        :param console_format: The format of messages logged to the console. Simple stdout is used if None specified
+        :param file_format: The format of messages logged to the file (A default is used if None specified)
+        :param file_level: min. Level of messages logged to the file
+        :param filename: The file the logs are written to
+        """
+        super().__init__(name, console_level)  # can be initialized with any level
+
+        # Logging to a file
+        if filename:
+            # Default format used if no file format provided
+            if file_format is None:
+                file_format = '%(asctime)s :: %(levelname)s :: %(message)s'
```

Comment:
Not necessary but one thing you could consider:

pull this out into a constant at the top, like

```python
DEFAULT_FILE_FORMAT = '%(asctime)s :: %(levelname)s :: %(message)s'
```
and then in the definition of __init__ you'd have as a default parameter, `file_format=DEFAULT_FILE_FORMAT`.

What you have is also good.